### PR TITLE
vkquake: 1.22.3 → 1.31.0

### LINKED ISSUE #329318: Update request: vkquake 1.22.3 → 1.31.0
- Package name: vkquake
- Latest released version: 1.22.3
- Current version on the unstable channel: 1.22.3
- Current version on the stable/release channel: 1.22.3

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@ylh

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by poperigby**
https://github.com/Novum/vkQuake/releases/tag/1.31.0

## Description of changes

I updated vkQuake from 1.22.3 to 1.31.0. I converted it to use Meson and Ninja instead of Make, as that's the new recommended way to build it.

Closes #329318

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Gliczy — APPROVED** [review]
Works as expected. Thanks for updating this!

**Comment by lavenderdotpet** [comment]
yooooo hope this gets merged

File: pkgs/games/quakespasm/vulkan.nix [hunk 1]
```diff
@@ -1,30 +1,52 @@
...
 
-  sourceRoot = "${src.name}/Quake";
-
   nativeBuildInputs = [
     makeWrapper
+    glslang
+    meson
+    ninja
+    pkg-config
     vulkan-headers
```
Comment (SuperSandro2000) [inline pkgs/games/quakespasm/vulkan.nix:39]: Aren't glslang/vulkan-headers supposed to go in buildInputs?

File: pkgs/games/quakespasm/vulkan.nix [hunk 2]
```diff
@@ -34,19 +56,16 @@ stdenv.mkDerivation rec {
...
     NIX_CFLAGS_COMPILE = "-Wno-error=unused-but-set-variable";
   };
 
   postFixup = ''
+    cp vkquake "$out/bin"
     wrapProgram $out/bin/vkquake \
       --prefix LD_LIBRARY_PATH : ${vulkan-loader}/lib
   '';
 
-  enableParallelBuilding = true;
```
Comment (SuperSandro2000) [inline pkgs/games/quakespasm/vulkan.nix:48]: Why did you disable this? does it no longer work?

Comment (poperigby) [reply pkgs/games/quakespasm/vulkan.nix:48]: `enableParallelBuilding` is set to true by default when using Meson: https://ryantm.github.io/nixpkgs/hooks/meson/

**Comment by poperigby** [comment]
Also, can I add myself as a maintainer?

**Comment by Gliczy** [comment]
The title for the commit adding you to the maintainers list should be `maintainers: add poperigby`.

**Comment by poperigby** [comment]
> The title for the commit adding you to the maintainers list should be `maintainers: add poperigby`.

Thanks for letting me know. Fixed.

File: pkgs/games/quakespasm/vulkan.nix [hunk 3]
```diff
@@ -34,19 +54,16 @@ stdenv.mkDerivation rec {
...
-  makeFlags = [ "prefix=$(out) bindir=$(out)/bin" ];
-
   env = lib.optionalAttrs stdenv.isDarwin {
     NIX_CFLAGS_COMPILE = "-Wno-error=unused-but-set-variable";
   };
 
   postFixup = ''
+    cp vkquake "$out/bin"
     wrapProgram $out/bin/vkquake \
       --prefix LD_LIBRARY_PATH : ${vulkan-loader}/lib
```
Comment (SuperSandro2000) [inline pkgs/games/quakespasm/vulkan.nix:64]:
```suggestion
    patchelf $out/bin/vkquake \
      --add-rpath ${lib.makeLibraryPath [ vulkan-loader ]}
```
we can avoid setting LD_LIBRARY_PATH

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 329327` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vkquake</li>
  </ul>
</details>

**Pull request merged**